analysers/analyser_merge_charging_station_FR.py

```diff
@@ -20,33 +20,147 @@
 ##                                                                       ##
 ###########################################################################
 
+import unittest
 from modules.OsmoseTranslation import T_
 from .Analyser_Merge import Analyser_Merge_Point, Source, CSV, Load_XY, Conflate, Select, Mapping
+import re
 
 
 class Analyser_Merge_Charging_station_FR(Analyser_Merge_Point):
+    def remove_trailing_zeros(input_string):
+        """
+        Removes dot and zeros at the end of a floating number typed in string
+        """
+        return str(input_string).replace('.00', '').replace('.0', '')
+
+    def map_wikidata_from_operator(self, str):
+
+        return str
+
+    @staticmethod
+    def is_float(str):
+        """
+        Returns True if the input string can be converted to a float, False otherwise.
+        """
+        pattern = r"^-?\d+(\.\d+)?$"
+        return bool(re.match(pattern, str))
+
+    def socket_output_find_correspondances(self, power: str):
+        """
+           convert the number of Watts to kiloWatts
+           output example:
+           - "400 kW"
+           - "7 kW"
+           - "50.6 kW"
+        """
+
+        power = power.replace(',', '.')
+
+        # remove extremely high values or the ones containing letters
+        max_output_kw = 1999
+        detection_watts = 2000
+        # we take an upper limit of 400 kW
+        max_kw = 401
+
+        # values under the max expected kW are used as is,
+        # upper values are Watts, which we should divide by 1000 to get kW
+
+        if not power:
+            return ''
+        if re.search(r"[a-zA-Z]+", power):
+            return ''
+
+        if not self.is_float(power) and int(power) > detection_watts:
+            power = int(power) / 1000
+
+        if not self.is_float(power) or float(power) < 1:
+            return ''
+
+        # clean the values of power
+        string_power = str(round(float(power), 2))
+        cleaned_power = (self.remove_trailing_zeros(string_power))
+        if float(power) < max_output_kw:
+            return '{0} kW'.format(cleaned_power)
+
+        else:
+            if float(power) > (max_kw * 1000):
+                return ''
+        if "." in str(cleaned_power):
+            if '.0' in str(cleaned_power) or '.00' in str(cleaned_power):
```

Comment:
Be aware that if .00 is in the string, .0 is also in the string, so the second check is unneeded

Reply:
yup, i removed that part and changed the function who checks for trailing zeros